src/component/axis/AxisBuilder.ts

```diff
@@ -809,6 +824,8 @@ function buildAxisLabel(
         transformGroup.add(textEl);
         textEl.updateTransform();
 
+        textEl.zlevel = labelItem.level;
```

Comment:
`zlevel` is configured by developers. Each `zlevel` represents an individual canvas. Instead, we can use `z2` to set the painting order of each element inside each component. So perhaps we can add this level to the `z2` in https://github.com/apache/echarts/blob/7d33cfb062d7035c5c3bce61bcb5a269d0762daf/src/component/axis/AxisBuilder.ts#L784

Another problem seems only `time` axis has this `level` property. We can either check if `level` exists or use a default `0` value.